src/dreyfus_index.erl

```diff
@@ -388,18 +389,18 @@ verify_index_exists(Options) ->
     Sig = dreyfus_util:get_value_from_options(<<"sig">>, Options),
     case couch_db:open_int(DbName, []) of
         {ok, Db} ->
-            IndexExist0 = verify_index(Db, DDocId, IdxName, Sig),
-            couch_db:close(Db),
-            IndexExist0;
-        _ ->
-            false
-    end.
-
-verify_index(Db, DDocId, IdxName, Sig) ->
-    case couch_db:open_doc(Db, DDocId, []) of
-        {ok, DDoc} ->
-            {ok, #index{sig=Sig0}} = dreyfus_index:design_doc_to_index(DDoc, IdxName),
-            Sig == Sig0;
+            try
+                {ok, DDoc} = couch_db:open_doc(Db, DDocId, []),
+                {ok, #index{sig=Sig0}} = dreyfus_index:design_doc_to_index(DDoc, IdxName),
+                Sig == Sig0
+            catch _:_ ->
+                couch_log:warning("Something wrong happened, please check it. "
```

Comment:
I think instead of `couch_log:warning` returning false is enough.  Also, a log contains thousands of messages, and when writing smth to log, you should specify the content what is the problem, and where it is coming from; otherwise it would be very difficult to debut it.